lib/v2/bilibili/userChannel.js

```diff
@@ -4,66 +4,74 @@ const cache = require('./cache');
 const utils = require('./utils');
 
 const notFoundData = {
-    title: '此 bilibili 频道不存在',
+  title: '此 bilibili 频道不存在',
 };
 
 module.exports = async (ctx) => {
-    const uid = parseInt(ctx.params.uid);
-    const sid = parseInt(ctx.params.sid);
-    const disableEmbed = ctx.params.disableEmbed;
+  const uid = parseInt(ctx.params.uid);
+  const sid = parseInt(ctx.params.sid);
+  const disableEmbed = ctx.params.disableEmbed;
 
-    const link = `https://space.bilibili.com/${uid}/channel/seriesdetail?sid=${sid}`;
+  const link = `https://space.bilibili.com/${uid}/channel/seriesdetail?sid=${sid}`;
 
-    // 获取频道信息
-    const channelInfoLink = `https://api.bilibili.com/x/series/series?series_id=${sid}`;
-    const channelInfo = await ctx.cache.tryGet(channelInfoLink, async () => {
-        const response = await got(channelInfoLink, {
-            headers: {
-                Referer: link,
-            },
-        });
-        // 频道不存在时返回 null
-        return response.data.data;
+  // 获取频道信息
+  const channelInfoLink = `https://api.bilibili.com/x/series/series?series_id=${sid}`;
+  const channelInfo = await ctx.cache.tryGet(channelInfoLink, async () => {
+    const response = await got(channelInfoLink, {
+      headers: {
+        Referer: link,
+      },
     });
+    // 频道不存在时返回 null
+    return response.data.data;
+  });
 
-    if (!channelInfo) {
-        ctx.state.data = notFoundData;
-        return;
-    }
-    const [userName, face] = await cache.getUsernameAndFaceFromUID(ctx, uid);
-    const host = `https://api.bilibili.com/x/series/archives?mid=${uid}&series_id=${sid}&only_normal=true&sort=desc&pn=1&ps=10`;
+  if (!channelInfo) {
+    ctx.state.data = notFoundData;
+    return;
+  }
 
-    const response = await got(host, {
-        headers: {
-            Referer: link,
-        },
-    });
+  const {
+    meta: { total },
+  } = channelInfo;
+  const pageSize = total || 100;
```

Comment:
This 100 should be replaced with the result of `const limit = parseInt(ctx.query.limit) || 100`, because the limit is a common parameter.

https://docs.rsshub.app/en/parameter.html#limit-entries